DMCompiler/DM/Builders/DMExpressionBuilder.cs

```diff
@@ -351,6 +351,8 @@ public enum ScopeMode {
                 return new Args(identifier.Location);
             case "__TYPE__":
                 return new ProcOwnerType(identifier.Location);
+            case "__IMPLIED_TYPE__":
+                return new ImpliedType(identifier.Location, inferredPath);
```

Comment:
This should give `new ConstantPath(identifier.Location, dmObject, inferredPath.Value);` instead of creating a new type of expression and re-implementing a bunch of behavior.